sdk/core/azure-core/inc/azure/core/http/curl/curl connection_pool.hpp

```diff
@@ -27,6 +27,77 @@ namespace Azure { namespace Core { namespace Test {
 
 namespace Azure { namespace Core { namespace Http {
 
+  /**
+   * @brief The available options to set libcurl ssl options.
+   *
+   * @remark The SDK will map the enum option to libcurl's specific option. See more info here:
+   * https://curl.haxx.se/libcurl/c/CURLOPT_SSL_OPTIONS.html
+   *
+   */
+  enum class CurlTransportSSLOptions
+  {
+    Default,
+    AllowBeast,
+    NoRevoke,
+    NoPartialchain,
```

Comment:
I don't think winhttp has this capability.

Reply:
Can we avoid adding functionality/capability too eagerly and only add them after they are explicitly requested by the customer with an actual scenario?

Since these are public, I recommend we start with the smallest set required.

Reply:
That's what we are doing.
The number of options for curl are much more!  
I am adding only the ones requested by storage team in the original issue.  The thing is that, for setting `CURLSSLOPT_NO_REVOKE`, I don't feel right creating a different approach (like changing default value, or changing the name) for the original option.  I think we have to just map to how the original is without trying to enhance it.